tests/php/ORM/DataObjectTest/MockDynamicAssignmentDBField.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace SilverStripe\ORM\Tests\DataObjectTest;
+
+use SilverStripe\ORM\FieldType\DBBoolean;
+use SilverStripe\ORM\FieldType\DBField;
+
+/**
+ * This is a fake DB field specifically design to test dynamic value assignment. You can set `scalarValueOnly` in
+ * the constructor. You can control whetever the field will try to do a dynamic assignment by specifing
+ * `$dynamicAssignment` in nthe consturctor.
+ *
+ * If the field is set to false, it will try to do a plain assignment. This is so you can save the initial value no
+ * matter what. If the field is set to true, it will try to do a dynamic assignment.
+ */
+class MockDynamicAssignmentDBField extends DBBoolean
+{
+
+    private $scalarOnly;
+    private $dynamicAssignment;
+
+    /**
+     * @param $name
+     * @param $scalarOnly Whether our fake field should be scalar only
+     * @param $dynamicAssigment Wheter our fake field will try to do a dynamic assignement
```

Comment:
This docblock is missing types, there's a typo in `$dynamicAssignment`, `Whether` and `assignment`

Reply:
Done.